src/common/misc.c

```diff
@@ -773,3 +773,108 @@ daos_hlc2timestamp(uint64_t hlc, time_t *ts)
 	*ts = tspec.tv_sec;
 	return 0;
 }
+
+/** Find requested number of unused bits (neither set it @used or @reserved */
+int
+daos_find_bits(uint64_t *used, uint64_t *reserved, int bmap_sz, int bits_min, int *bits)
+{
+	int	nr_saved;
+	int	at_saved;
+	int	nr;
+	int	at;
+	int	i;
+	int	j;
+
+	nr = nr_saved = 0;
+	at = at_saved = -1;
+
+	for (i = 0; i < bmap_sz; i++) {
+		uint64_t free_bits = ~used[i];
+
+		if (reserved)
+			free_bits &= ~reserved[i];
+
+		if (free_bits == 0) { /* no space in the current int64 */
+			if (nr > nr_saved) {
+				nr_saved = nr;
+				at_saved = at;
+			}
+			nr = 0;
+			at = -1;
+			continue;
+		}
+
+		j = ffsll(free_bits);
```

Comment:
why only do this for first one?  Is it expensive?  Or is it expected that all of the bits will be set after that point?

There are also some compiler intrinsics such as __builtin_ctzll

Reply:
Sorry for late reply, this function mostly copied from Liang's original admem codes, ffsll() accept a uint64. it might be fine for new vea bitmap allocator since mostly it only find one bit.